tasks/fr/insee.py

```diff
@@ -0,0 +1,333 @@
+            weight=0,
+            targets={geom: GEOM_REF}
+        )
+
+        return OrderedDict([
+            ('the_geom', geom),
+            ('dcomiris', geomref)
+        ])
+
+
+class OutputAreas(TableTask):
+
+    def requires(self):
+        return {
+            'geom_columns': OutputAreaColumns(),
+            'data': ImportOutputAreas(),
+        }
+
+    def version(self):
+        return 2
+
+    def timespan(self):
+        return 2013
+
+    def columns(self):
+        input_ = self.input()
+        cols = OrderedDict()
+        cols.update(input_['geom_columns'])
+        return cols
+
+    def populate(self):
+        session = current_session()
+        session.execute('INSERT INTO {output} '
+                        'SELECT ST_MakeValid(wkb_geometry), DCOMIRIS '
+                        'FROM {input}'.format(
+                            output=self.output().table,
+                            input=self.input()['data'].table,
+                        ))
+
+
+class FranceCensus(TableTask):
+
+    table_theme = Parameter()
+    resolution = Parameter()
+
+    def version(self):
+        return 7
+
+    def timespan(self):
+        return '2012'
+
+    def requires(self):
+        requirements = {
+            'data': RawFRData(table_theme=self.table_theme, resolution=self.resolution),
+            'meta': FrenchColumns(table_theme=self.table_theme),
+            'geometa': OutputAreaColumns(),
+        }
+        return requirements
+
+    def columns(self):
+        cols = OrderedDict()
+        cols['IRIS'] = self.input()['geometa']['dcomiris']
+        cols.update(self.input()['meta'])
+        return cols
+
+    def populate(self):
+        session = current_session()
+
+        column_targets = self.columns()
+        colnames = ', '.join(column_targets.keys())
+        colnames_typed = ','.join(['{}::{}'.format(colname, ct.get(session).type)
+                              for colname, ct in column_targets.iteritems()])
+        session.execute('INSERT INTO {output} ({ids}) '
+                        'SELECT {ids_typed} '
+                        'FROM {input} '.format(
+                            ids=colnames,
+                            ids_typed=colnames_typed,
+                            output=self.output().table,
+                            input=self.input()['data'].table
+                        ))
+
+
+class AllGeomsThemesTables(WrapperTask):
+    def requires(self):
+        topics = ['population', 'housing', 'education', 'household', 'employment']
+        for resolution in ('iris', 'iris_overseas'):
+            for table_theme in topics:
+                yield FranceCensus(table_theme=table_theme, resolution=resolution)
+
+
+class AllGeometries(WrapperTask):
+
+    def requires(self):
+        geom_types = ('iris', 'overseas_iris')
+        for geom in geom_types:
+            yield OutputAreas(geom=geom)
```

Comment:
This is currently broken because `OutputAreas` no longer takes a `geom` parameter.

If `OutputAreas` is going to generate all sorts of other geoms, it should be added back, even if it is ignored for now.